components/include/cn_array_utils.h

```diff
@@ -2,10 +2,54 @@
 #define CN_ARRAY_UTILS_H_
 
 /*$
+predicate (map<u64,u8>) ArrayBlock_u8 (pointer p, u64 e)
```

Comment:
Even though they are relatively simple, can you please add at least minimal documentation to these predicates, such that non-experts can understand their purpose and why they are written the way they are? Think TA2 engineers who are not FM experts. This will be also useful for any future AI assistant learning.

E.g. `ArrayBlock iterates over an array pointed to by pointer p for up to e iterations... CN requires these predicates because...`